sdk/Eversign/Client.php

```diff
@@ -446,15 +460,16 @@ public function downloadRawDocumentToPath(Document $document, $path) {
          return $this->downloadDocumentToPath($document, $path, 0, Config::DOCUMENT_RAW_URL);
      }
 
-     private function downloadDocumentToPath(Document $document, $path, $auditTrail = 0, $type = Config::DOCUMENT_FINAL_URL) {
+    private function downloadDocumentToPath(Document $document, $path, $auditTrail = 0, $documentId, $type = Config::DOCUMENT_FINAL_URL) {
```

Comment:
We should not introduce the variable in the middle of function variables (it would break existing calls in code).

What we can do (a common approach) when we need to extend function capabilities:
- You put new input variable at the end of the function (so order is not corrupted for existing function calls)
- You assign a default value to it (so existing calls don't have to provide it and they won't break)
- Which default value to assign? The one which provides same functionality as before (before that variable was even there)

End result would look like:
private function downloadDocumentToPath(Document $document, $path, $auditTrail = 0, $type = Config::DOCUMENT_FINAL_URL, $documentId = "DefaultValueThatReflectsOldSate") {

You can test it afterwards